### PR TITLE
[external assets] report_asset_check endpoint

### PR DESCRIPTION
**Pull request by alangenfeld**
adds a rest endpoint for reporting runless asset check evaluations

## How I Tested These Changes

added test

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17133** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17133" target="_blank">Graphite</a> 
    * **PR #17136** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17136" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/17136?utm_source=stack-comment).

File: python_modules/dagster-webserver/dagster_webserver/external_assets.py [hunk 1]
```diff
@@ -11,6 +16,28 @@
 )
 
 
+def _asset_key_from_request(key: str, request: Request, json_body):
+    check.invariant(key == "asset_key")  #
```
Comment (schrockn) [inline python_modules/dagster-webserver/dagster_webserver/external_assets.py:20]: rm `#`

File: python_modules/dagster-webserver/dagster_webserver/external_assets.py [hunk 2]
```diff
@@ -114,15 +120,105 @@ async def handle_report_asset_materialization_request(
...
+    else:
+        return JSONResponse(
+            {
+                "error": (
+                    f"Unhandled content type {body_content_type}, expect no body or"
+                    " application/json"
+                ),
+            },
+            status_code=400,
+        )
```
Comment (schrockn) [inline python_modules/dagster-webserver/dagster_webserver/external_assets.py:145]: seems like there is some code consolidation to be had with `handle_report_asset_materialization_request` as this is pretty much pure [copypasta](https://app.graphite.dev/macros/dagster-io)

**Review by schrockn — CHANGES_REQUESTED** [review]
Can you do a pass to consolidate some of the duplicated code?

File: python_modules/dagster-webserver/dagster_webserver/external_assets.py (hunk 2 above)
Comment (alangenfeld) [reply python_modules/dagster-webserver/dagster_webserver/external_assets.py:145]: the remaining copy-pasta-esque code have control flow `return` statements, so factoring them is non trivial and i decided that the few lines of simple code duplication were preferable.

**Review by schrockn — APPROVED** [review]
ok deferring your judgment on that

**Pull request merged**